### ISSUE TITLE
error : Generation failed: System.MissingMethodException

### ISSUE THREAD

**Issue by mfkl** [unoplatform/uno#6137]
## Current behavior

```
2021-06-02T09:01:52.1351770Z "D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj" (Build target) (30:5) ->
2021-06-02T09:01:52.1352439Z (_UnoSourceGenerator target) -> 
2021-06-02T09:01:52.1353752Z   MSBUILD : error : Generation failed: System.MissingMethodException: Method not found: 'System.Collections.Immutable.ImmutableList`1<!!0> System.Collections.Immutable.ImmutableList.ToImmutableList(Builder<!!0>)'. [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1355065Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.UpdateOperation.ApplyImpl(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1356240Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemOperation.Apply(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1357480Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemList.ComputeItems(LazyItemList lazyItemList, ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1358620Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemList.GetItemData(ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1359638Z MSBUILD : error :    at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1360563Z MSBUILD : error :    at System.Linq.Buffer`1..ctor(IEnumerable`1 source) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1361500Z MSBUILD : error :    at System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1362407Z MSBUILD : error :    at Microsoft.Build.Evaluation.Evaluator`4.Evaluate() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1364329Z MSBUILD : error :    at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1366943Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1368314Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1369656Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1371831Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1373522Z MSBUILD : error :    at Uno.SourceGeneration.Host.ProjectLoader.LoadProjectDetails(BuildEnvironment environment, Dictionary`2 globalProperties) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1375285Z MSBUILD : error :    at Uno.SourceGeneration.Host.SourceGeneratorEngine.Generate() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1376482Z MSBUILD : error :    at Uno.SourceGeneration.Host.Program.RunGeneration(String generatedFilesOutputPath, String binlogOutputPath, BuildEnvironment environment, Boolean enableConsole) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1377810Z   C:\Users\VssAdministrator\.nuget\packages\uno.sourcegenerationtasks\1.32.0-dev.297\build\netstandard1.0\Uno.SourceGenerationTasks.targets(127,2): error : Generation failed, error code 3 [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1378479Z 
2021-06-02T09:01:52.1378677Z 
2021-06-02T09:01:52.1379235Z "D:\a\1\s\src\LibVLCSharp.sln" (Build target) (1) ->
2021-06-02T09:01:52.1379897Z "D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj" (default target) (30) ->
2021-06-02T09:01:52.1380606Z "D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj" (Build target) (30:6) ->
2021-06-02T09:01:52.1381647Z   MSBUILD : error : Generation failed: System.MissingMethodException: Method not found: 'System.Collections.Immutable.ImmutableList`1<!!0> System.Collections.Immutable.ImmutableList.ToImmutableList(Builder<!!0>)'. [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1382929Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.UpdateOperation.ApplyImpl(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1384439Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemOperation.Apply(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1385597Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemList.ComputeItems(LazyItemList lazyItemList, ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1386740Z MSBUILD : error :    at Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemList.GetItemData(ImmutableHashSet`1 globsToIgnore) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1387754Z MSBUILD : error :    at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1389327Z MSBUILD : error :    at System.Linq.Buffer`1..ctor(IEnumerable`1 source) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1390254Z MSBUILD : error :    at System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1391142Z MSBUILD : error :    at Microsoft.Build.Evaluation.Evaluator`4.Evaluate() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1392808Z MSBUILD : error :    at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1395183Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1396558Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1397786Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1399048Z MSBUILD : error :    at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1400346Z MSBUILD : error :    at Uno.SourceGeneration.Host.ProjectLoader.LoadProjectDetails(BuildEnvironment environment, Dictionary`2 globalProperties) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1402134Z MSBUILD : error :    at Uno.SourceGeneration.Host.SourceGeneratorEngine.Generate() [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1404484Z MSBUILD : error :    at Uno.SourceGeneration.Host.Program.RunGeneration(String generatedFilesOutputPath, String binlogOutputPath, BuildEnvironment environment, Boolean enableConsole) [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
2021-06-02T09:01:52.1406394Z   C:\Users\VssAdministrator\.nuget\packages\uno.sourcegenerationtasks\1.32.0-dev.297\build\netstandard1.0\Uno.SourceGenerationTasks.targets(127,2): error : Generation failed, error code 3 [D:\a\1\s\src\LibVLCSharp.Uno\LibVLCSharp.Uno.csproj]
```

## Expected behavior

No build error.

## Minimal reproduction of the problem with instructions

That's the trouble. I cannot reproduce locally...

The CI has started failing a couple days ago, on Windows only (macOS seems happy). We have no updated anything with regard to Uno.

Our Uno dependency is quite behind `Uno.UI (>= 2.0.527)`, but I guess this should keep working fine? My quick guess is that a downward nuget dependency is updating while it should not?

We intend on updating to latest Uno to bring support to libvlcsharp, but in the meantime the CI is kinda broken :-)

If this is expected, feel free to close and I guess we will have to upgrade to newer Uno faster than later.

## Environment
```
Nuget Package: Uno.UI 2.0.527

Package Version(s): 

Affected platform(s):
- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [x] Windows
- [x] Build tasks
```

**Comment by mfkl** [comment]
> To keep building for 2.x, you can include an explicit reference to Uno.SourceGenerationTasks 3.0.2 or later.

Thank you! That fixed it.

> Note that most developers are now using Uno 3.0+, which is binary incompatible with Uno 2.x.

Yes, we plan to move to Uno 3 hopefully soon 👍 

Cheers

**Comment by jeromelaban** [comment]
Thanks for the update :) Let us know if you have any trouble updating to Uno 3!

**Comment by vincentcastagna** [comment]
@jeromelaban  this issue still happens on **3.8.0-dev.439** with Stable Version of VS 2019 16.10.0 

You can test this on https://github.com/nventive/UnoApplicationTemplate by building solution in this particular version of VS .

**Comment by jeromelaban** [comment]
The issue is fixed in latest 3.8 and 3.7, there are no additional changes we can make.

**Comment by jeromelaban** [comment]
Let's track this in another issue, it's not the same problem.
